src/cc_mm.c

```diff
@@ -34,10 +34,8 @@ _cc_alloc(size_t size, const char *name, int line)
 {
     void *p;
 
-    ASSERT(size != 0);
-
     p = malloc(size);
-    if (p == NULL) {
+    if (p == NULL && size != 0) {
```

Comment:
That was actually intentional; according to the [C99 Standard](http://cs.nyu.edu/courses/spring13/CSCI-GA.2110-001/downloads/C99.pdf):
"The  order  and  contiguity  of  storage  allocated  by  successive  calls  to  the calloc, malloc, and realloc functions is unspecified.  The pointer returned if the allocation succeeds is suitably aligned so that it may be assigned to a pointer to any type of object and then used to access such an object or an array of such objects in the space allocated (until  the  space  is  explicitly  deallocated).   The  lifetime  of  an  allocated  object  extends from the allocation until the deallocation.  Each such allocation shall yield a pointer to an object disjoint from any other object.  The pointer returned points to the start (lowest byte address)  of  the  allocated  space.   If  the  space  cannot  be  allocated,  a  null  pointer  is returned.   **If  the  size  of  the  space  requested  is  zero,  the  behavior  is  implementation-defined:  either  a  null  pointer  is  returned,  or  the  behavior  is  as  if  the  size  were  some nonzero value, except that the returned pointer shall not be used to access an object.**"